src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,165 @@
+package duke;
+
+import java.io.IOException;
+
+import java.util.regex.Pattern;
+
+/**
+ * Parses the response taken by user from Ui.
+ */
+public class Parser {
+
+	private static Pattern CHECKSTRING = Pattern.compile("-?\\d+");
+
+	private boolean isDone;
+
+	private Storage storage;
+	private TaskList tasks;
+
+	enum Update {
```

Comment:
I feel that `Update` could perhaps be changed into `TaskUpdater` instead. Converting it from a verb to a noun.